lib/smack/SmackInstGenerator.cpp

```diff
@@ -145,39 +145,165 @@ void SmackInstGenerator::processInstruction(llvm::Instruction& inst) {
 }
 
 void SmackInstGenerator::processIntrinsicCall(llvm::IntrinsicInst* ii) {
+  static const auto overApproximate = [this](llvm::IntrinsicInst* ii) {
+    errs() << "warning: over-approximate call to intrinsic: "
+      << ii->getCalledFunction()->getName() << "\n";
+    emit(rep->call(ii->getCalledFunction(), *ii));
+  };
+  static const auto assignFuncApp = [&ii, this]
+    (bool flag,
+     std::string funcNameBase,
+     std::initializer_list<std::string> funcNameArgs,
+     std::list<const Expr*> args) {
+      if (flag)
```

Comment:
Why do you have this `flag` here, instead of calling `overApproximate` directly from `stmtMap`?
Just use `overApproximate` directly in `stmtMap`. It will make the code cleaner.

Reply:
The more I think about this, to me it seems we should remove `assignFuncApp`.

Reply:
And replace it with?

Reply:
> Why do you have this `flag` here, instead of calling `overApproximate` directly from `stmtMap`?
> Just use `overApproximate` directly in `stmtMap`. It will make the code cleaner.

How can `overApproximate` be used in `stmtMap`?